### PR TITLE
Add e2e-metal-ipi CI job

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml [hunk 1]
```diff
@@ -1200,6 +1200,85 @@ presubmits:
         secret:
           secretName: sentry-dsn
     trigger: (?m)^/test( | .* )e2e-azure-upi,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
+    branches:
+    - master
+    context: ci/prow/e2e-baremetalds
```
Comment (russellb) [inline ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml]: Can we make the name that's visible to users something like `e2e-metal-ipi` ?  I don't think `baremetalds` is going to be meaningful to most people

**Comment by stbenjam** [comment]
/hold

Still need to address naming, but at least job is running

**Comment by stbenjam** [comment]
/hold cancel

This should be good to go.

**Comment by crawford** [comment]
The tests are still failing and while digging into the exact failure, I wasn't able to find the typical set of artifacts that are saved when creating a cluster. Did I miss something?

**Comment by stbenjam** [comment]
> The tests are still failing and while digging into the exact failure, I wasn't able to find the typical set of artifacts that are saved when creating a cluster. Did I miss something?

We can't use the step-registry gather step and have to write one, it's in progress. We don't want it to hold up this job. 

The failure's expected at this point, baremetal IPI deploys have been flakey over the last week - we're digging into it, but even getting the jobs in place in the state they are can be valuable for preventing regressions on IPv6, etc. We don't expect it to be blocking on anything until it's stable.

**Comment by crawford** [comment]
I see, this uses the "new" step/chain/workflow mechanism. I need to familiarize myself with the new paradigm. This looks fine.

/approve
/lgtm

**Pull request merged**